src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -386,6 +392,131 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
     );
   };
 
+  /**
+   * Applies coloring to point traces in a scatterplot. Does this by splitting the data into multiple traces each with a solid
+   * color, which is much faster than using Plotly's native color ramping. Also enforces a maximum number of points
+   * per trace, which significantly speeds up Plotly renders.
```

Comment:
Nit: minor concern that a scientist looking very closely at the points will notice the binning and therefore inaccurate visualization. Just something to anticipate, maybe unlikely that it comes up.

Reply:
That's fair. I imagine if you spread the color ramp ends VERY far apart (like, 100X past the ends of the color ramp) it might be visible? At some point we'd be running into the limitations of RGBA itself anyways, since there's a max of 256 values (and therefore a max of 256 relevant subsampled colors) per channel.